cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -0,0 +1,162 @@
+/*
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dao.watersupply;
+
+import cwms.cda.data.dao.location.kind.LocationUtil;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.data.dto.watersupply.PumpType;
+import cwms.cda.data.dto.watersupply.WaterSupplyPump;
+import cwms.cda.data.dto.watersupply.WaterUser;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import usace.cwms.db.dao.util.OracleTypeMap;
+import usace.cwms.db.jooq.codegen.udt.records.LOOKUP_TYPE_OBJ_T;
+import usace.cwms.db.jooq.codegen.udt.records.LOOKUP_TYPE_TAB_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_CONTRACT_OBJ_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_CONTRACT_REF_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_CONTRACT_TAB_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_OBJ_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_TAB_T;
+
+
+public final class WaterSupplyUtils {
```

Comment:
Cant this whole class be package scoped?

Reply:
Yes, will update that in the controller PR